internal/services/notification/db/db.go

```diff
@@ -83,12 +83,30 @@ func mustSingleRow[T any](s []*T) (*T, error) {
 	return s[0], nil
 }
 
-func (d *DB) GetRunWebhookDeliveriesAfterSequence(tx *sql.Tx, afterSequence uint64, deliveryStatus types.DeliveryStatus, limit int) ([]*types.RunWebhookDelivery, error) {
+func (d *DB) GetRunWebhookDeliveriesAfterSequence(tx *sql.Tx, afterSequence uint64, projectID string, deliveryStatusFilter []types.DeliveryStatus, limit int, sortDirection types.SortDirection) ([]*types.RunWebhookDelivery, error) {
```

Comment:
Use a new function to filter by project and use the old one for the global run webhooks (you could also remove deliveryStatus arg if not used.)